src/app/3d/qgs3dviewsmanager.cpp

```diff
@@ -0,0 +1,148 @@
+/***************************************************************************
+  qgs3dviewsmanager.cpp
+  --------------------------------------
+  Date                 : December 2021
+  Copyright            : (C) 2021 by Belgacem Nedjima
+  Email                : gb underscore nedjima at esi dot dz
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgs3dviewsmanager.h"
+
+#include "qgisapp.h"
+#include "qgs3dmapcanvasdockwidget.h"
+#include "qgsnewnamedialog.h"
+#include "qgs3dmapcanvas.h"
+
+Qgs3DViewsManager::Qgs3DViewsManager( QWidget *parent, Qt::WindowFlags f )
+  : QDialog( parent, f )
+{
+  setupUi( this );
+  m3DViewsListView->setModel( &mListModel );
+
+  m3DViewsListView->setEditTriggers( QAbstractItemView::NoEditTriggers );
+  m3DViewsListView->setSelectionMode( QAbstractItemView::SingleSelection );
+
+  connect( mOpenButton, &QToolButton::clicked, this, &Qgs3DViewsManager::openClicked );
+  connect( mDuplicateButton, &QToolButton::clicked, this, &Qgs3DViewsManager::duplicateClicked );
+  connect( mRemoveButton, &QToolButton::clicked, this, &Qgs3DViewsManager::removeClicked );
+  connect( mRenameButton, &QToolButton::clicked, this, &Qgs3DViewsManager::renameClicked );
+}
+
+void Qgs3DViewsManager::openClicked()
+{
+  if ( m3DViewsListView->selectionModel()->selectedRows().isEmpty() )
+    return;
+
+  QString viewName = mListModel.stringList()[ m3DViewsListView->selectionModel()->selectedRows().at( 0 ).row() ];
```

Comment:
You could take this directly from the data instead:

```suggestion
  QString viewName = m3DViewsListView->selectionModel()->selectedIndexes().at( 0 ).data( Qt::DisplayRole );
```